lib/matplotlib/lines.py

```diff
@@ -978,16 +989,26 @@ def set_color(self, color):
 
     def set_drawstyle(self, drawstyle):
         """
-        Set the drawstyle of the plot
+        Set the drawstyle of the plot.
 
-        'default' connects the points with lines. The steps variants
-        produce step-plots. 'steps' is equivalent to 'steps-pre' and
-        is maintained for backward-compatibility.
+        The drawstyle determines how the points are connected.
 
         Parameters
         ----------
         drawstyle : {'default', 'steps', 'steps-pre', 'steps-mid', \
 'steps-post'}
+            Possible values:
+
+            - 'default': The points are connected with straight lines.
+            - 'steps-pre', 'steps-mid', 'steps-post': The points are
+              connected with step-like lines, i.e. horizontal lines with
+              vertical steps.
+              The postfix pre/mid/post determines the position of the vertical
+              step (directly after the left point / halfway between the
```

Comment:
having "pre" correspond to "*after* the left point" sounds a bit confusing to me; I guess the more intuitive picture is "before the horizontal line" (and so on).

Reply:
I was stumbling over that already. For some reason I'm thinking more in "is the y-value of the line determined by the point to the left or to the right" not in where is the step (probably because I'm usually more interested in the value of the line than in the position of the step). Of course, "which is the relevant point" and "where is the step" are inverse to each other concerning pre/post :slightly_frowning_face:. I've now extended the text to include both views. I hope that adds too the clarity and is not too verbose.